doc/rm/otbn_coding_style.md

```diff
@@ -0,0 +1,232 @@
+---
+title: "OTBN Assembly Style Guide"
+---
+
+Where possible, OTBN assembly should follow the same principles as the [RISC-V assembly style guide]({{< relref "asm_coding_style" >}}).
+This guide describes additional OTBN-specific guidelines, and places where OTBN assembly can or should diverge from the RISC-V style guidelines.
+
+## General Advice
+
+### Register Names
+
+OTBN registers cannot be referred to by ABI names due to toolchain limitations.
```

Comment:
This isn't quite phrased right: there's no ABI for OTBN, so they don't *have* ABI names...